util/middleware.js

```diff
@@ -0,0 +1,85 @@
+if (typeof (fetch) === 'undefined') require('isomorphic-fetch')
+
+/**
+ * This method builds the options object for call to the fetch method.
+ * @param {string} accessToken If non-null, a bearer Authorization header will be added with the specified token.
+ * @param {string} apiKey If non-null, an x-api-key header will be added with the specified key.
+ * @param {string} method The HTTP method to execute.
+ * @param {*} options Extra options to pass to fetch.
+ */
+export function getSecureFetchOptions (accessToken, apiKey, method = 'get', options = {}) {
+  const headers = {
+    // JSON request bodies only.
+    'Content-Type': 'application/json'
+  }
+  if (apiKey) {
+    headers['x-api-key'] = apiKey
+  }
+  if (accessToken) {
+    headers.Authorization = `Bearer ${accessToken}`
+  }
+
+  return {
+    method,
+    mode: 'cors', // Middleware is at a different URL.
+    headers,
+    ...options
+  }
+}
+
+/**
+ * This convenience method wraps a fetch call to the specified URL
+ * with the token and api key added (if provided) to the HTTP request header,
+ * and wraps the response by adding the success/error status of the call.
+ * @param {string} url The URL to call.
+ * @param {string} accessToken If non-null, the Authorization token to add to request header.
+ * @param {string} apiKey If non-null, the API key to add to the Authorization header.
+ * @param {string} method The HTTP method to execute.
+ * @param {*} options Extra options to pass to fetch.
+ */
+export async function secureFetch (url, accessToken, apiKey, method = 'get', options = {}) {
```

Comment:
This should be merged with the secureFetch method contained in #6.

Reply:
And the snake is biting its tail...